pkg/sql/sqlstats/insights/integration/integration_test: TestInsightsIntegrationForContention failed

pkg/sql/sqlstats/insights/integration/integration_test.TestInsightsIntegrationForContention [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12358050?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12358050?buildTab=artifacts#/) on master @ [c2ffc96f93f7a0dac9799788a042f962148e8af0](https://github.com/cockroachdb/cockroach/commits/c2ffc96f93f7a0dac9799788a042f962148e8af0):


```
=== RUN   TestInsightsIntegrationForContention
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/8db7e2fbdb77cb910d7e66552998132e/logTestInsightsIntegrationForContention3904684906
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:124: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
    insights_test.go:755: condition failed to evaluate within 45s: sql: Scan error on column index 4, name "database_name": converting NULL to string is unsupported
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/8db7e2fbdb77cb910d7e66552998132e/logTestInsightsIntegrationForContention3904684906
--- FAIL: TestInsightsIntegrationForContention (48.91s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cluster-observability
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInsightsIntegrationForContention.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-32749

--- a/pkg/sql/sqlstats/insights/integration/insights_test.go
+++ b/pkg/sql/sqlstats/insights/integration/insights_test.go
@@ -758,9 +758,9 @@ func TestInsightsIntegrationForContention(t *testing.T) {
 		COALESCE(insight.contention, 0::INTERVAL)::FLOAT,
 		COALESCE(sum(txn_contention.contention_duration), 0::INTERVAL)::FLOAT AS durationMs,
 		COALESCE(txn_contention.schema_name, ''::STRING)::STRING AS schema_name,
-		txn_contention.database_name,
-		txn_contention.table_name,
-		txn_contention.index_name,
+		COALESCE(txn_contention.database_name, ''::STRING)::STRING AS database_name,
+		COALESCE(txn_contention.table_name, ''::STRING)::STRING AS table_name,
+		COALESCE(txn_contention.index_name, ''::STRING)::STRING AS index_name,
 		encode(txn_contention.waiting_txn_fingerprint_id, 'hex') AS waiting_txn_fingerprint_id
 		FROM crdb_internal.cluster_execution_insights insight
 		left join crdb_internal.transaction_contention_events txn_contention on  insight.stmt_id = txn_contention.waiting_stmt_id

